### PR TITLE
MGDOBR-928: Display cloud provider/region information of a Bridge

### PR DESCRIPTION
**Pull request by ppadti**
JIRA:  https://issues.redhat.com/browse/MGDOBR-928

Steps to test it (on chrome):

- Open the drawer of smart event instance details. 
- It will display the Cloud Provider and Cloud Region information of the instance.
- Also try to create a new Instance and check the drawer for the same.
- If the providers are not available then it will display Cloud Provider and Cloud Region  as "Not available".

### CODE REVIEW

File: src/app/Instance/InstanceDetails/InstanceDetails.tsx [hunk 1]
```diff
@@ -91,6 +95,26 @@ export const InstanceDetails = ({
               {instance.published_at && formatDate(instance.published_at)}
             </DescriptionListDescription>
           </DescriptionListGroup>
+          <DescriptionListGroup>
+            <DescriptionListTerm>
+              {t("instance.cloudProvider")}
+            </DescriptionListTerm>
+            <DescriptionListDescription data-ouia-component-id="instance-details-cloud-provider">
+              {updatedCloudProvider?.label == undefined
+                ? "Not available"
```
Comment (vpellegrino) [inline src/app/Instance/InstanceDetails/InstanceDetails.tsx:104]: probably this string can be translated too

File: src/app/Instance/InstanceDetails/InstanceDetails.tsx [hunk 2]
```diff
@@ -91,6 +95,26 @@ export const InstanceDetails = ({
...
+                : updatedCloudProvider.label}
+            </DescriptionListDescription>
+          </DescriptionListGroup>
+          <DescriptionListGroup>
+            <DescriptionListTerm>
+              {t("instance.cloudRegion")}
+            </DescriptionListTerm>
+            <DescriptionListDescription data-ouia-component-id="instance-details-cloud-region">
+              {updatedCloudRegion?.label == undefined
+                ? "Not available"
```
Comment (vpellegrino) [inline src/app/Instance/InstanceDetails/InstanceDetails.tsx:114]: probably this string can be translated too

File: src/app/Instance/InstanceDetails/InstanceDetails.tsx [hunk 3]
```diff
@@ -91,6 +95,26 @@ export const InstanceDetails = ({
...
             </DescriptionListDescription>
           </DescriptionListGroup>
+          <DescriptionListGroup>
+            <DescriptionListTerm>
+              {t("instance.cloudProvider")}
+            </DescriptionListTerm>
+            <DescriptionListDescription data-ouia-component-id="instance-details-cloud-provider">
+              {updatedCloudProvider?.label == undefined
+                ? "Not available"
+                : updatedCloudProvider.label}
```
Comment (vpellegrino) [inline src/app/Instance/InstanceDetails/InstanceDetails.tsx:105]:
```suggestion
              {updatedCloudProvider?.label ?? "Not available"}
```

File: src/app/Instance/InstanceDetails/InstanceDetails.tsx [hunk 4]
```diff
@@ -91,6 +95,26 @@ export const InstanceDetails = ({
...
+            </DescriptionListDescription>
+          </DescriptionListGroup>
+          <DescriptionListGroup>
+            <DescriptionListTerm>
+              {t("instance.cloudRegion")}
+            </DescriptionListTerm>
+            <DescriptionListDescription data-ouia-component-id="instance-details-cloud-region">
+              {updatedCloudRegion?.label == undefined
+                ? "Not available"
+                : updatedCloudRegion.label}
```
Comment (vpellegrino) [inline src/app/Instance/InstanceDetails/InstanceDetails.tsx:115]:
```suggestion
              {updatedCloudRegion?.label ?? "Not available"}
```

File: src/types/CloudProviders.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+const cloudProviderOptions: CloudProviderDetails[] = [
+  {
+    value: "aws",
+    label: "Amazon Web Services",
+    region: [{ value: "us-east-1", label: "US East, N. Virginia" }],
+  },
+];
+
+export const getUpdatedCloudProviders = (
```
Comment (vpellegrino) [inline src/types/CloudProviders.ts:21]: Probably in this context, this function name makes not much sense. WDYT about `getCloudProviderAndRegionForInstance` or `getCloudProviderAndRegionByInstance`?

File: src/types/CloudProviders.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  instance: BridgeResponse
+): {
+  updatedCloudProvider: CloudProviderDetails | undefined;
+  updatedCloudRegion: RegionDetails | undefined;
```
Comment (vpellegrino) [inline src/types/CloudProviders.ts:25]:
```suggestion
  cloudProvider: CloudProviderDetails | undefined;
  cloudRegion: RegionDetails | undefined;
```

**Review by vpellegrino — CHANGES_REQUESTED** [review]
Thank you for your PR @ppadti :rocket: 
The code overall looks good. I left a few minor suggestions.

File: src/types/CloudProviders.ts (hunk 5 above)
Comment (ppadti) [reply src/types/CloudProviders.ts:21]: Yes, Okay
Thanks @vpellegrino

**Review by jomarko — COMMENTED** [review]
Thank you for a PR @ppadti . The only think I think we miss is more coverage. I prepared a PR https://github.com/ppadti/sandbox-ui/pull/1 that is updating existing cypress test. Once coverage is increased (you can simply merge my PR if you are ok with code), I can approve.

**Review by vpellegrino — APPROVED** [review]

**Comment by ppadti** [comment]
> @ppadti any updates on my comment ^ ?

Yes, Its done.

**Review by jomarko — APPROVED** [review]
Thank you for your work @ppadti

File: src/app/Instance/InstanceDetails/InstanceDetails.tsx [hunk 7]
```diff
@@ -91,6 +95,22 @@ export const InstanceDetails = ({
               {instance.published_at && formatDate(instance.published_at)}
             </DescriptionListDescription>
           </DescriptionListGroup>
+          <DescriptionListGroup>
+            <DescriptionListTerm>
+              {t("instance.cloudProvider")}
+            </DescriptionListTerm>
+            <DescriptionListDescription data-ouia-component-id="instance-details-cloud-provider">
+              {cloudProvider?.label ?? "Not available"}
```
Comment (kelvah) [inline src/app/Instance/InstanceDetails/InstanceDetails.tsx:103]: Like suggested by Valentino, we should leave no untranslated strings inside templates.

Could you please add a translation for `Not available`?

You'll have to add a new key/value pair inside the dictionary located at `locales/en/openbridge.json`, inside the `common` list I would say.

**Review by kelvah — COMMENTED** [review]
Nice work @ppadti :ok_hand: 

Left you only one request.

File: src/app/Instance/InstanceDetails/InstanceDetails.tsx (hunk 7 above)
Comment (ppadti) [reply src/app/Instance/InstanceDetails/InstanceDetails.tsx]: The changes have been made.

**Review by kelvah — APPROVED** [review]
LGTM :+1:

**Pull request merged**